jsondir/Main.hs

```diff
@@ -6,14 +6,39 @@ module Main where
 import Data.Aeson
 import Control.Monad
 import System.Environment
+import System.FilePath
 import Data.JSON.Directory
 import qualified Data.ByteString.Lazy.Char8 as B
+import System.Process
+import qualified Data.ByteString.Char8 as BS
+import qualified Data.Text as Text
+import Data.List
+
+parseArgs :: [String] -> ([Rule], [FilePath])
+parseArgs ("--rule":rest) = case rest of
+    pat:filter:rest ->
+        let
+            (rules, roots) = parseArgs rest
+            rule = Rule
+                { predicate = isSuffixOf pat
+                , jsonKey   = Text.pack . takeBaseName
+                , parser    = \fp -> do
+                    s <- readFile fp
+                    r <- readCreateProcess (shell filter) s
+                    pure $ idecodeStrict (BS.pack r)
```

Comment:
Might be good to treat non-zero exit codes as failures and return `IError`.

Reply:
`readCreateProcess` throws an `IOError` on non-zero exit codes. I now catch that and turn it into an `IResult` so that we get the JSON path in the error result.